src/VirtoCommerce.CatalogModule.Data/Handlers/IndexCategoryChangedEventHandler.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Hangfire;
+using VirtoCommerce.CatalogModule.Core.Events;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Events;
+using VirtoCommerce.SearchModule.Core.Model;
+using VirtoCommerce.SearchModule.Core.Services;
+
+namespace VirtoCommerce.CatalogModule.Data.Handlers
+{
+    public class IndexCategoryChangedEventHandler : IEventHandler<CategoryChangedEvent>
+    {
+        private readonly IIndexingManager _indexingManager;
+
+        public IndexCategoryChangedEventHandler(IIndexingManager indexingManager)
+        {
+            _indexingManager = indexingManager;
+        }
+
+        public Task Handle(CategoryChangedEvent message)
+        {
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            var indexCategoryIds = message.ChangedEntries.Where(x => (x.EntryState == EntryState.Modified || x.EntryState == EntryState.Added) && x.OldEntry.Id != null)
```

Comment:
Why we van't handle deleted product as well as updated?

Reply:
we can